src/main/java/org/opensrp/web/rest/LocationResource.java

```diff
@@ -357,18 +363,25 @@ public void setLocationService(PhysicalLocationService locationService) {
 
 	/**
 	 * This methods provides an API endpoint that searches for all structure ids
+	 * sorted by server version ascending
 	 *
-	 * @return A list of structure Ids
+	 * @param serverVersion serverVersion using to filter by
+	 * @return A list of structure Ids and last server version
 	 */
 	@RequestMapping(value = "/findStructureIds", method = RequestMethod.GET, produces = {
 			MediaType.APPLICATION_JSON_VALUE })
-	public ResponseEntity<String> findIds() {
+	public ResponseEntity<Identifier> findIds(
+			@RequestParam(value = SERVER_VERSION)  long serverVersion) {
 
 		try {
-			return new ResponseEntity<>(
-					gson.toJson(locationService.findAllStructureIds()), RestUtils.getJSONUTF8Headers(), HttpStatus.OK);
+
+			Pair<List<String>, Long> structureIdsPair = locationService.findAllStructureIds(serverVersion, DEFAULT_GET_ALL_IDS_LIMIT);
+			Identifier identifiers = new Identifier();
+			identifiers.setIdentifiers(structureIdsPair.getLeft());
+			identifiers.setLastServerVersion(structureIdsPair.getRight());
+			return new ResponseEntity<>(identifiers, RestUtils.getJSONUTF8Headers(), HttpStatus.OK);
 		} catch (Exception e) {
```

Comment:
can we catch something more specific here?

Reply:
This catches all possible errors preventing leaking of stack-trace. This can be removed the global error handler is configured https://github.com/OpenSRP/opensrp-server-web/issues/245